l10n_it_fatturapa_out/wizard/efattura.py

```diff
@@ -110,7 +110,7 @@ def format_quantity(line):
         def get_id_fiscale_iva(partner, prefer_fiscalcode=False):
             id_paese = partner.country_id.code
             if partner.vat:
-                if id_paese == "IT" and partner.vat.startswith("IT"):
+                if partner.vat.startswith(id_paese):
```

Comment:
L'IdPaese non è il prefisso della vat. Sono due cose diverse.

Ci sono vat che iniziano per EL, per XI, per EU. Non ha senso togliere un prefisso a certe vat estere e ad altre no. Lo SdI controlla solo quelle italiane e vuole un numero, per questo togliamo IT.

La fattura di cortesia mi pare alleghi la stampa standard e in quella IdPaese non c'è. Quindi non c'è ripetizione.

Solo il pulsante/link preview dell'XML renderizza tutto il blocco IdFiscaleIVA e concatena IdPaese con IdCodice.

Di fatto ciò crea situazioni come GREL123456789 o GBXI123456789 o USEU123456789, e di conseguenza anche GBGB123456789 o DEDE123456789.  Siccome non esiste nessuna garanzia che il numero strippato del prefisso sia univoco, non ha senso toglierlo.

Non si può mappare entrambe le vat XI123456789 e GB123456789 inviando allo SdI la stessa identica stanza 
```
<IdFiscaleIVA>
    <IdPaese>GB</IdPaese>
    <IdCodice>123456789</IdCodice>
</IdFiscaleIVA>
```
solo per migliorare l'estetica del preview dell'XML.

Né ha senso strippare il secondo caso e non il primo, inviando:
```
<IdFiscaleIVA>
    <IdPaese>GB</IdPaese>
    <IdCodice>XI123456789</IdCodice>
</IdFiscaleIVA>
```
per la vat XI123456789  e
```
<IdFiscaleIVA>
    <IdPaese>GB</IdPaese>
    <IdCodice>123456789</IdCodice>
</IdFiscaleIVA>
```
per la vat GB123456789.

Sia GB che XI sono parte integrante dell'IdCodice, indipendentemente che IdPaese sia GB.

Abbiamo 4 casistiche:
- IdPaese e il prefisso vat coincidono (es. DE DE)
- IdPaese e il prefisso vat non coincidono (es. GR EL)
- IdPaese è uno, ma i prefissi sono due e uno coincide (GB con GB/XI)
- per il prefisso EU, IdPaese non solo non coincide ma non è nemmeno determinato (potrebbe essere US, CH, ecc.)

La modifica proposta stripperebbe il prefisso vat per il primo caso e per metà del terzo, e non per gli altri.

Reply:
@TheMule71 L'unica casistica ambigua è quando abbiamo un'azienda che risiede ed opera in un paese, ad esempio Germania, ma ha una partita iva ed ad esempio Francese. Solo in in questo caso non dobbiamo strippare, per tutti gli altri casi il codice della PR li copre tutti e permette di avere una preview corretta delle fatture emesse.

Reply:
Ribadisco non vedo la logica nel strippare (in alcuni casi)  un prefisso che  è parte del  VAT no. solo per evitare di correggere un foglio di stile che di fatto fa la cosa sbagliata se nel visualizzare il VAT no. ci mette davanti l'IdPaese , che non c'entra nulla (al punto che una una codifica diversa).

Se si toglie il prefisso quando è uguale a IdPaese, nel tuo caso ti troveresti come VAT no. nel preview DEFR123456789.

Il fix corretto è togliere l'IdPaese e ottenere FR123456789.  Negli altri casi che ho elencato otterresti DE123456789 (corretto), GB123456789 (corretto), XI123456789 (corretto), EL123456789 (corretto), EU123456789 (corretto).

Togli IdPaese dal XLTS e funziona per tutte le casistiche. Per di più IdPaese non è un'informazione particolarmente utile perché è una ripetizione della country del partner che è già indicata nell'indirizzo.

Reply:
>>Se si toglie il prefisso quando è uguale a IdPaese, nel tuo caso ti troveresti come VAT no. nel preview DEFR123456789.

Si hai ragione in questo caso è effettivamente come hai detto tu. Però il concetto rimane sempre lo stesso se il country code coincide con le prime 2 lettere della VAT questo deve essere strippato. Solo nel caso DEFR.... allora non andrebbe prefissato l'IdPaese, ma siccome questo dipende dal XSLT, che non lo vogliamo toccare, possiamo almeno coprire tutte le altre casistiche.